libs/build_netcdf.sh

```diff
@@ -63,7 +63,7 @@ LDFLAGS1="-L$HDF5_ROOT/lib"
 LDFLAGS2=$(cat $HDF5_ROOT/lib/libhdf5.settings | grep AM_LDFLAGS | cut -d: -f2)
 [[ $enable_pnetcdf =~ [yYtT] ]] && LDFLAGS4="-L$PNETCDF_ROOT/lib"
 if [[ ${STACK_netcdf_shared:-} != [yYtT] ]]; then
-  LDFLAGS1+=" -lhdf5_hl -lhdf5"
+  LDFLAGS1+=" -lhdf5 -lhdf5_hl -lz"
```

Comment:
I have a feeling that some compilers read linkers left to right and some other way.
Is that true?

Reply:
That could be, yes, however I was never able to independently verify that.

Reply:
At any rate, I have reverted both the hdf5 and netcdf library builds to what is found in the develop branch of noaa-emc/hpc-stack.